Intersect.Client/Entities/Player.cs

```diff
@@ -733,33 +733,95 @@ private void SellInputBoxOkay(object sender, EventArgs e)
             PacketSender.SendSellItem((int)((InputBox)sender).UserData, 1);
         }
 
-        //bank
-        public void TryDepositItem(int inventorySlotIndex, int bankSlotIndex = -1)
+        /// <summary>
+        /// Tries to deposit items into the bank.
+        /// </summary>
+        /// <param name="inventorySlotIndex">The slot of the item in the player's inventory.</param>
+        /// <param name="bankSlotIndex">The slot of the bank to deposit the item into.</param>
+        /// <param name="contextMenu">Indicates if the action was triggered by a context menu.</param>
+        /// <param name="shiftKey">Indicates if the action was triggered by the shift key</param>
+        public void TryDepositItem(int inventorySlotIndex,
+            int bankSlotIndex = -1,
+            bool contextMenu = false,
+            bool shiftKey = false)
```

Comment:
This shoud have been called prompt as I said on Discord, the name of the key does not convey what the parameter is for

Reply:
shiftKey is used for the alternative shift + double click logic !

Reply:
Ok but the functionality is showing or not showing a prompt which is relevant in this part of the code.

The key to do it is irrelevant at this level of abstraction, and when the client and server inventory abstractions are merged it will be even less relevant to the code.